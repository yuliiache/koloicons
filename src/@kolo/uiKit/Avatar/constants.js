export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
