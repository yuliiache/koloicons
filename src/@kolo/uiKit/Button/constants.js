export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
