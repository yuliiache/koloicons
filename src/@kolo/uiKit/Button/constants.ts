export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
