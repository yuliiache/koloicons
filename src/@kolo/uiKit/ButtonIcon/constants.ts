export const BUTTON_ICON_TEST_ID = 'buttonIconComponent';

export enum ButtonIconType {
  ROUND = 'round',
  SQUARE = 'square',
}
