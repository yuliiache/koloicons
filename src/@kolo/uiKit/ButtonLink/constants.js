export const BUTTONLINK_TEST_ID = 'buttonLinkComponent';

