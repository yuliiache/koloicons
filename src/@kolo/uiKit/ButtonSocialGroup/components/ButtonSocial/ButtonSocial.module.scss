@import "@kolo/design";

.buttonSocial {
  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  padding: 12px 16px;

  font-family: $mainFontFamily;
  font-weight: 500;
  font-size: 0.7rem;
  text-align: center;
  color: $secFontColor;

  background-color: $WhiteBG;
  border: none;
  cursor: pointer;

  transition: box-shadow 0.2s ease-out;

  &:hover {
    color: $BlueColor;
  }

  &:focus {
    position: relative;
    outline: 4px solid rgba($color: $BlueColor, $alpha: 0.2);
    z-index: 1;
  }

  &Icon {
    display: inline-flex;

    & svg {
      width: 24px;
      height: 24px;
    }
  }

  &Text {
    margin-right: 16px;
  }

  &Disconnected {
    &:hover {
      color: $RedColor;

      &Text,
      &Icon {
        opacity: 1;
      }
    }

    &Text,
    &Icon {
      opacity: 0.7;
    }
  }
}
