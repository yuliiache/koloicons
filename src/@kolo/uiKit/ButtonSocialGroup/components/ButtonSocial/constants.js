export const BUTTON_SOCIAL_TEST_ID = 'buttonSocialComponent';

export const BUTTON_SOCIAL_TEXT_TEST_ID = `${BUTTON_SOCIAL_TEST_ID}Text`;
