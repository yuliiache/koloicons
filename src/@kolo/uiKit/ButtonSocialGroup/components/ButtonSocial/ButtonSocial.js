import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import {BUTTON_SOCIAL_TEST_ID, BUTTON_SOCIAL_TEXT_TEST_ID} from './constants';

import styles from './ButtonSocial.module.scss';

export const ButtonSocial = ({text, shouldShowStatus, isConnected, renderIcon, onClick}) => {
  const isDisconnected = shouldShowStatus && !isConnected;

  const buttonClasses = classNames(styles.buttonSocial, {
    [styles.buttonSocialDisconnected]: isDisconnected,
  });

  let buttonText = text;

  if (shouldShowStatus) {
    buttonText = isConnected ? `Connect ${text}` : `Disconnect ${text}`;
  }

  return (
    <button
      data-testid={BUTTON_SOCIAL_TEST_ID}
      className={classNames(buttonClasses)}
      type="button"
      onClick={onClick}
    >
      <span
        className={styles.buttonSocialText}
        data-testid={BUTTON_SOCIAL_TEXT_TEST_ID}
      >
        {buttonText}
      </span>
      <i className={styles.buttonSocialIcon}>{renderIcon && renderIcon()}</i>
    </button>
  );
};

ButtonSocial.propTypes = {
  text: PropTypes.string,
  shouldShowStatus: PropTypes.bool,
  isConnected: PropTypes.bool,
  renderIcon: PropTypes.func,
  onClick: PropTypes.func,
};
