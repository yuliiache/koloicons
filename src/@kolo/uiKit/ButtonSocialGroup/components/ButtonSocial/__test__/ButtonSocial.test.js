import React from 'react';
import {cleanup, render, screen, waitFor} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import {ButtonSocial} from '../ButtonSocial';
import {BUTTON_SOCIAL_TEST_ID} from '../constants';

describe('ButtonSocial component', () => {
  afterEach(() => {
    cleanup();
  });

  describe('snapshot', () => {
    const renderIconFunc = () => <span>MockedIcon</span>;

    it('should render snapshot without icon render function', () => {
      const props = {
        text: 'Facebook',
        onClick: jest.fn(),
      };

      const {asFragment} = render(<ButtonSocial {...props} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should render succesfully without status prefix without connect status', () => {
      const props = {
        text: 'Facebook',
        renderIcon: renderIconFunc,
        onClick: jest.fn(),
      };

      const {asFragment} = render(<ButtonSocial {...props} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should render succesfully with status prefix with connect status', () => {
      const props = {
        text: 'Facebook',
        isConnected: true,
        renderIcon: renderIconFunc,
        shouldShowStatus: true,
        onClick: jest.fn(),
      };

      const {asFragment} = render(<ButtonSocial {...props} />);

      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('should has text without status prefix when shouldShowStatus property is not provided', () => {
    const buttonText = 'Facebook';
    const props = {
      text: buttonText,
      onClick: jest.fn(),
    };

    render(<ButtonSocial {...props} />);

    const buttonSocialTextElement = screen.getByText(buttonText);

    expect(buttonSocialTextElement).toBeInTheDocument();
  });

  it('should has appropriate prefix in text', () => {
    const buttonText = 'Facebook';
    const props = {
      text: buttonText,
      shouldShowStatus: true,
      isConnected: true,
      onClick: jest.fn(),
    };

    render(<ButtonSocial {...props} />);

    const buttonSocialTextElement = screen.getByText(`Connect ${buttonText}`);

    expect(buttonSocialTextElement).toBeInTheDocument();
  });

  it('should has appropriate prefix in text when button is disconnected', () => {
    const buttonText = 'Facebook';

    const props = {
      text: buttonText,
      shouldShowStatus: true,
      isConnected: false,
      onClick: jest.fn(),
    };

    render(<ButtonSocial {...props} />);

    const buttonSocialTextElement = screen.getByText(`Disconnect ${buttonText}`);

    expect(buttonSocialTextElement).toBeInTheDocument();
  });

  it('should call onClick function', async () => {
    const onClick = jest.fn();
    const props = {
      text: 'test',
      onClick,
    };

    render(<ButtonSocial {...props} />);

    const buttonSocialElement = screen.getByTestId(BUTTON_SOCIAL_TEST_ID);

    userEvent.click(buttonSocialElement);

    await waitFor(() => expect(onClick).toHaveBeenCalled());
  });
});
