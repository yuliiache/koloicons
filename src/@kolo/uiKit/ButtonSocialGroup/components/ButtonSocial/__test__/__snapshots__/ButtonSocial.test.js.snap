// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSocial component snapshot should render snapshot without icon render function 1`] = `
<DocumentFragment>
  <button
    class="buttonSocial"
    data-testid="buttonSocialComponent"
    type="button"
  >
    <span
      class="buttonSocialText"
      data-testid="buttonSocialComponentText"
    >
      Facebook
    </span>
    <i
      class="buttonSocialIcon"
    />
  </button>
</DocumentFragment>
`;

exports[`ButtonSocial component snapshot should render succesfully with status prefix with connect status 1`] = `
<DocumentFragment>
  <button
    class="buttonSocial"
    data-testid="buttonSocialComponent"
    type="button"
  >
    <span
      class="buttonSocialText"
      data-testid="buttonSocialComponentText"
    >
      Connect Facebook
    </span>
    <i
      class="buttonSocialIcon"
    >
      <span>
        MockedIcon
      </span>
    </i>
  </button>
</DocumentFragment>
`;

exports[`ButtonSocial component snapshot should render succesfully without status prefix without connect status 1`] = `
<DocumentFragment>
  <button
    class="buttonSocial"
    data-testid="buttonSocialComponent"
    type="button"
  >
    <span
      class="buttonSocialText"
      data-testid="buttonSocialComponentText"
    >
      Facebook
    </span>
    <i
      class="buttonSocialIcon"
    >
      <span>
        MockedIcon
      </span>
    </i>
  </button>
</DocumentFragment>
`;
